protocol/src/types.rs

```diff
@@ -218,6 +218,57 @@ fn tlv_sanity_check(tlv: &LldpTlv, expected_type: TlvType) -> Result<()> {
     }
 }
 
+fn hex2str(data: &[u8]) -> String {
+    if data.is_empty() {
+        "null".to_string()
+    } else {
+        data.iter()
+            .map(|a| format!("{a:02x}"))
+            .collect::<Vec<String>>()
+            .join(":")
+    }
+}
+
+#[derive(
+    Clone,
+    Debug,
+    Hash,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Deserialize,
+    JsonSchema,
+    Serialize,
+)]
+pub enum NetworkAddress {
+    IpAddr(IpAddr),
+    IEEE802(Vec<u8>),
```

Comment:
Is this a variable length value, or is it always a 6 byte MAC? I'm having a bit of trouble finding the LLDP spec for this.

Reply:
"Address 6" means "802 (includes all 802 media plus Ethernet "canonical format")", so it isn't restricted to MAC addresses.  I don't have any idea if anybody uses anything other than MAC addresses, but I wasn't comfortable making that a restriction.

Reply:
Can you add a link to the spec you're going off in a comment somewhere.

Reply:
Done.